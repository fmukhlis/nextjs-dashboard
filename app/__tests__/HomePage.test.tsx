import { render, screen } from "@testing-library/react";
import Home from "../page";

describe("Homepage", () => {
  it("renders a login button", () => {
    render(<Home />);
    const loginButton = screen.getByRole("link", { name: /log in/i });
    expect(loginButton).toBeInTheDocument();
  });

  // Snapshot
  it("renders Homepage unchanged", () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });
});
