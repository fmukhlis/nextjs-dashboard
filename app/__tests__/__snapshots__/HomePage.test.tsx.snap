// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Homepage renders Homepage unchanged 1`] = `
<div>
  <main
    class="flex min-h-screen flex-col p-6"
  >
    <div
      class="flex h-20 shrink-0 items-end rounded-lg bg-blue-500 p-4 md:h-52"
    >
      <div
        class="className flex flex-row items-center leading-none text-white"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-earth h-12 w-12 rotate-45"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21.54 15H17a2 2 0 0 0-2 2v4.54"
          />
          <path
            d="M7 3.34V5a3 3 0 0 0 3 3a2 2 0 0 1 2 2c0 1.1.9 2 2 2a2 2 0 0 0 2-2c0-1.1.9-2 2-2h3.17"
          />
          <path
            d="M11 21.95V18a2 2 0 0 0-2-2a2 2 0 0 1-2-2v-1a2 2 0 0 0-2-2H2.05"
          />
          <circle
            cx="12"
            cy="12"
            r="10"
          />
        </svg>
        <p
          class="text-[44px] tracking-wider ml-3"
        >
          Acme
        </p>
      </div>
    </div>
    <div
      class="mt-4 flex grow flex-col gap-4 md:flex-row"
    >
      <div
        class="flex flex-col justify-center gap-6 rounded-lg bg-gray-50 px-6 py-10 md:w-2/5 md:px-20"
      >
        <p
          class="text-xl text-gray-800 md:text-3xl md:leading-normal"
        >
          <strong>
            Welcome to Acme.
          </strong>
           This is the example for the
           
          <a
            class="text-blue-500"
            href="https://nextjs.org/learn/"
          >
            Next.js Learn Course
          </a>
          , brought to you by Vercel.
        </p>
        <div
          class="relative w-0 h-0 border-l-[15px] border-r-[15px] border-b-[26px] border-l-transparent border-r-transparent border-b-black"
        />
        <a
          class="className tracking-widest flex items-center gap-5 self-start rounded-lg bg-blue-500 px-6 py-3 text-sm font-medium text-white transition-colors hover:bg-blue-400 md:text-base"
          href="/login"
        >
          <span>
            Log in
          </span>
           
          <svg
            aria-hidden="true"
            class="lucide lucide-arrow-big-right w-5 md:w-6 fill-white"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11 9a1 1 0 0 0 1-1V5.061a1 1 0 0 1 1.811-.75l6.836 6.836a1.207 1.207 0 0 1 0 1.707l-6.836 6.835a1 1 0 0 1-1.811-.75V16a1 1 0 0 0-1-1H5a1 1 0 0 1-1-1v-4a1 1 0 0 1 1-1z"
            />
          </svg>
        </a>
      </div>
      <div
        class="flex items-center justify-center p-6 md:w-3/5 md:px-28 md:py-12"
      >
        <img
          alt="Screenshots of the dashboard project showing desktop version"
          class="hidden md:block"
          data-nimg="1"
          decoding="async"
          height="760"
          loading="lazy"
          src="/_next/image?url=%2Fhero-desktop.png&w=2048&q=75"
          srcset="/_next/image?url=%2Fhero-desktop.png&w=1080&q=75 1x, /_next/image?url=%2Fhero-desktop.png&w=2048&q=75 2x"
          style="color: transparent;"
          width="1000"
        />
        <img
          alt="Screenshots of the dashboard project showing mobile version"
          class="block md:hidden"
          data-nimg="1"
          decoding="async"
          height="620"
          loading="lazy"
          src="/_next/image?url=%2Fhero-mobile.png&w=1200&q=75"
          srcset="/_next/image?url=%2Fhero-mobile.png&w=640&q=75 1x, /_next/image?url=%2Fhero-mobile.png&w=1200&q=75 2x"
          style="color: transparent;"
          width="560"
        />
      </div>
    </div>
  </main>
</div>
`;
